modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2812,6 +2814,23 @@ protected void updateModelForComposedSchema(CodegenModel m, Schema schema, Map<S
         // end of code block for composed schema
     }
 
+    private void mergeProperties(Map<String, Schema> properties, Map<String, Schema> interfaceProperties) {
```

Comment:
can you please add a docstring explaining what this function does?

some comments in the code block below would be nice.

Reply:
Thank you for the review! I've updated and added some detail, which hopefully makes the addition clearer.

Thank you!